docs/proposals/local-cluster.md

```diff
@@ -0,0 +1,486 @@
+  VM for `minikube` may cause noticeable slowdown.
+* Local development environments and easy to use single-script demos
+  for Kubernetes-related projects. This use case may be covered to an
+  extent using `minikube` yet the previous points about an ability to
+  run such scripts on CI and resource usage are also valid for this
+  use case. DIND solutions provides the possibility of a more
+  lightweight demo for Docker users, see
+  [Virtlet demo](https://github.com/Mirantis/virtlet#virtlet-usage-demo)
+  as an example.
+* Support for other platforms besides Linux. You can run `minikube` on
+  your Mac (at the cost of having to run a VM for it) but you can't
+  run `hack/local-up-cluster.sh` there unless you make an effort to
+  run it in a container.
+* Support for remote docker engines. If you prefer lightweight laptop
+  for your work it would be nice to be able to utilize some powerful
+  remote box or a cloud instance for your Kubernetes work. You can
+  [build Kubernetes](https://github.com/kubernetes/kubernetes/tree/master/build#really-remote-docker-engine)
+  this way, but what about test clusters? Moreover, if you have a
+  habit of working from places with limited network speed, while
+  current k8s build container rsync functionality will save you some
+  traffic, you still have to pull back bulky binaries like
+  `hyperkube`.
+
+These use cases are actually already covered by
+[kubeadm-dind-cluster](https://github.com/Mirantis/kubeadm-dind-cluster). The
+plan is to adapt this project for inclusion in `kubeadm` repository.
+
+## Proposal
+
+The `local-cluster` tool consists of a docker image that contains
+Docker engine and main `local-cluster` binary and a small startup
+script that starts the control container and downloads proper version
+of `kubectl` to the user's machine. There are several image versions
+for supported versions of Kubernetes that include corresponding
+version of `hyperkube` and `kubeadm` binaries. Also there's 'bare'
+image that doesn't include `hyperkube` and `kubeadm` binaries and can
+only be used for building Kubernetes from source.
+
+The cluster itself consists of several docker containers used as
+Kubernetes nodes. The containers are created with `systemd` and
+`docker` running inside them, then `kubeadm` is used to install
+Kubernetes onto them.
+
+![local-cluster architecture](local-cluster.png)
+
+_Rationale:_ concerning the startup script and control container: we
+need docker image anyway, so why bother with building `local-cluster`
+binaries for different supported platforms? Also, with control
+container the cluster can still start successfully if the connection
+between user's machine and docker engine breaks. Concerning multiple
+image versions with different binaries included: the idea is to be
+able to start cluster on user's machine without downloading much data
+once the image is pulled.
+
+## Usage
+
+`local-cluster.sh` startup script defaults to start the most recent
```

Comment:
Ok, as my starter script idea seems to be not very good after all, let's provide `local-cluster-up` binaries for Linux and Mac, will update the doc.